### PR TITLE
Add worker constraints: 4 cores and 4GB RAM

### LINKED ISSUE #338: Improve constraints in bundles
The default kubernetes-core and CDK bundles have no constraints. As a result, the base instance on AWS is so small that it is barely possible to run Helm Tiller pod once nginx and the dashboard components are loaded. Not great from a UX perspective.  

It would make sense to add a constraint on the worker nodes to add 4 cores and 4GB RAM so users can start playing with the system, and/or to scale to at least 2 workers and keep the same sizing.

### PR DESCRIPTION
**Pull request by Cynerva**
Fixes #338 

I've tested the kubernetes-core bundle built from this. Will give canonical-kubernetes a go tomorrow.

### CODE REVIEW

**Comment by tvansteenburgh** [comment]
LGTM

File: fragments/k8s/core/bundle.yaml [hunk 1]
```diff
@@ -30,6 +30,7 @@ services:
     options:
       channel: 1.7/stable
     expose: true
+    constraints: "cores=4 mem=4G"
```
Comment (ktsakalozos) [inline fragments/k8s/core/bundle.yaml]: I did not know we needed this in both the machine description and application.

**Comment by ktsakalozos** [comment]
LGTM+1

File: fragments/k8s/core/bundle.yaml (hunk 1 above)
Comment (Cynerva) [inline fragments/k8s/core/bundle.yaml]: Me neither! After testing, though, I think we do.

If you don't have constraints on the machine, then the initial deploy will have worker on a 1-core machine.

If you don't have constraints on the application, then `juju get-constraints kubernetes-worker` returns blank. I think that means new units created with `juju add-unit` won't get the constraints, but I didn't confirm that.

Comment (ktsakalozos) [inline fragments/k8s/core/bundle.yaml]: Ah, makes sense now. Thanks

**Comment by Cynerva** [comment]
Tested canonical-kubernetes with this change, too. Ready to merge now.

**Pull request merged**